Fix #2345 - Make bump & unify

pmbrull opened the pull request:
### Describe your changes :
<!-- Explain what you have done & tag your assigned issue !-->
This PR fixes https://github.com/open-metadata/OpenMetadata/issues/2345 and https://github.com/open-metadata/OpenMetadata/issues/2342

We have:
1. Moved the `ingestion-core` recipes to the main `Makefile` with the `core` prefix
2. The `make core_bump_version_dev` now handles the virtualenv installation in a separate clean environment to avoid clashes
3. Updated the names of the recipes in the workflows and added information on how to install precommits when checkstyle fails.
4. Added a default `help` recipe that outputs the description of all the commands in a fun openmetadata-y color. It should also make it easier for us to keep adding/improving the recipes as we can now document them directly.
5. Updated make and phony information following Makefile best practices

Could we remove `env38` and `clean_env37` recipes? It might be more interesting to leave the Python installation and setup to more explanatory docs (https://github.com/open-metadata/OpenMetadata/issues/2294). IDK if we are using them somewhere.

Thanks 🙏  Happy to discuss

#
### Type of change :
<!-- You should choose 1 option and delete options that aren't relevant -->
- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:
<!-- add an x in [] if done, don't mark items that you didn't do !-->
- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
<!-- Please see the contributing guidelines and then add your reviewer(s) !-->
<!--- OpenMetadata community thanks you for explaining your changes in detail !-->
<!--- If you are unsure of people to review your work, you can add anyone of these developers :) !-->
<!--- Frontend: @shahsank3t, @darth-coder00, @Sachin-chaurasiya -->
<!--- Backend: @sureshms @harshach -->
<!--- Ingestion: @harshach @ayush-shah @pmbrull -->

pmbrull commented on .github/workflows/java-checkstyle.yml:
```diff
@@ -54,7 +54,10 @@ jobs:
           body: |
             **The Java checkstyle failed.**
 
-            Please run `mvn googleformatter:format@reformat-sources` in the root of your repository and commit the changes to this PR. You can also use [pre-commit](https://pre-commit.com/) to automate the Java code formatting.
```
I am not quite sure if we had in a single line for any specific reason

harshach commented on .github/workflows/openmetadata-ingestion-core-version-comment.yml:
```diff
@@ -37,4 +37,4 @@ jobs:
             **Schema Change Detected. Needs py-ingestion-core version bump**
 
-            Please run `make bump-version-dev` in the ingestion-core folder of your repository and commit the changes to _version.py in this PR. Please ignore if this has been handled already.
+            Please run `make core_bump_version_dev` in the ingestion-core folder of your repository and commit the changes to _version.py in this PR. Please ignore if this has been handled already.
```
@pmbrull looks great, since we are removing the Makefile from ingestion-core. Can we change the text above to run the command in root folder?

pmbrull replied:
Nice catch, thanks

pmbrull commented on Makefile:
```diff
@@ -1,57 +1,79 @@
-isort:
+## Checkstyle
+.PHONY: isort
+isort:  ## Run isort command to standardize Python imports
```
Hi @mithmatt, agree on merging `isort` and `black` into a single recipe. However, I'd not go through all the environment and installation steps. This is something to be primarily used during development, so the python setup would be already expected.

I personally run these recipes multiple times while coding, so installing etc. might be repeated unnecessary steps.

pmbrull replied:
Thanks for pointing this out, I've also merged the `check` recipe to run `isort` as well and updated the python checkstyle workflow

pmbrull commented:
> **The Python checkstyle failed.**
> 
> Please run `make black` in the root of your repository and commit the changes to this PR. You can also use [pre-commit](https://pre-commit.com/) to automate the Python code formatting.

Please ignore this comment. It is failing as I updated the `make` recipes, so the workflow is not finding the old `make black_check`. I updated the CI accordingly with the new recipes. Thanks!

harshach commented:
@pmbrull python-check style docker is failing

pmbrull commented:
> @pmbrull python-check style docker is failing

Hi, @harshach! It is complaining from `make: *** No rule to make target 'black_check'.  Stop.` as I updated the recipe name to `py_format_check` to include an `isort` validation as well. The workflow has been updated with the new name
